internal/credentials/docker.go

```diff
@@ -16,23 +15,35 @@ type DockerConfigJSON struct {
 	Auths AuthConfigs `json:"auths"`
 }
 
-// ExtractDockerAuth will return a username/password combination from a JSON-representation of a Docker config file.
-func ExtractDockerAuth(input []byte, host string) (string, string, error) {
+// ExtractAuthConfigs will return the Docker AuthConfigs from a JSON-representation of a Docker config file.
+func ExtractAuthConfigs(input []byte) (authConfigs AuthConfigs, err error) {
+	authConfigs = AuthConfigs{}
+
 	r := bytes.NewReader(input)
 
 	cf := configfile.New("")
 	if err := cf.LoadFromReader(r); err != nil {
-		return "", "", err
+		return nil, err
+	}
+
+	// convert from the CLI type to the API type
+	for host, conf := range cf.GetAuthConfigs() {
+		authConfigs[host] = types.AuthConfig(conf)
 	}
 
-	ac, ok := cf.GetAuthConfigs()[host]
+	return authConfigs, nil
+}
+
+// ExtractBasicAuthForHost will extract the basic auth info for a registry host from an AuthConfigs instance
+func ExtractBasicAuthForHost(authConfigs AuthConfigs, host string) (string, string, error) {
+	ac, ok := authConfigs[host]
```

Comment:
why not flip this around
```suggestion
	if ac, ok := authConfigs[host]; ok {
	  return ac.Username, ac.Password, nil
	}
	...
	return the error
```

Reply:
FYI this pattern was present in the original code. I have updated to use your suggestion.